builtin/printf_osh.py

```diff
@@ -310,7 +310,8 @@ def _Percent(
 
             if match.LooksLikeInteger(s):
```

Comment:
Hm changing the meaning of LooksLikeInteger may have some unintended consequences, since it is used in at least 2 other places

```
$ fgrep -n 'LooksLikeInteger(' */*.py
builtin/printf_osh.py:311:            if match.LooksLikeInteger(s):
builtin/process_osh.py:609:            if match.LooksLikeInteger(s):
...

osh/sh_expr_eval.py:1116:            if match.LooksLikeInteger(s):

```